### DISCUSSION TITLE
amqp10_client OTP27 compilation fails

### DISCUSSION THREAD

**Discussion by nuex** [rabbitmq/rabbitmq-server#12176] [Other]
When using the amqp10_client hexpm dependency in an Elixir 1.17.2-otp-27 project, compilation fails:

```
Error: "Could not compile dependency :amqp10_client"
```

Earlier output suggests a conflict with OTP27 `maybe` expression, which is now enabled by default:

```
===> Compiling src/amqp10_msg.erl failed
amqp10_msg.erl:41:16: syntax error before: '::'
amqp10_msg.erl:55:35: syntax error before: 'maybe'
amqp10_msg.erl:59:46: syntax error before: 'maybe'
amqp10_msg.erl:81:20: syntax error before: 'maybe'
amqp10_msg.erl:145:5: syntax error before: 'maybe'
```

Setting `ERL_COMPILER_OPTIONS="[{feature, maybe_expr, disable}]"` allows `mix compile` to succeed. Maybe the `-feature(maybe_expr, disable).` module attribute could be added to amqp10_msg so the client can be compiled with default settings?

**Comment by mkuratczyk** [comment]
Recent versions of RabbitMQ can be compiled with OTP27. I guess we should just push a new version of `amqp_client` to hex.pm - it's been quite some time and indeed the versions available on hex are not OTP27-compatible.
